tools/clang/lib/Parse/ParseDecl.cpp

```diff
@@ -1900,6 +1899,14 @@ Parser::DeclGroupPtrTy Parser::ParseDeclaration(unsigned Context,
     ProhibitAttributes(attrs);
     SingleDecl = ParseDeclarationStartingWithTemplate(Context, DeclEnd);
     break;
+    // HLSL Change Begin.
+  case tok::kw_export:
+    // Ignore export for now.
+    ConsumeToken();
```

Comment:
It's less of "it has no effect" than it is a case of "we ignore it". Good thing to clean up for HLSL 2017 though.